src/data_source/storage/sql/queries/state.rs

```diff
@@ -371,23 +416,34 @@ fn build_get_path_query<'q>(
 ) -> QueryResult<(QueryBuilder<'q>, String)> {
     let mut query = QueryBuilder::default();
     let mut traversal_path = traversal_path.into_iter().map(|x| x as i32);
-    let created = query.bind(created)?;
 
     // We iterate through the path vector skipping the first element after each iteration
     let len = traversal_path.len();
     let mut sub_queries = Vec::new();
     for _ in 0..=len {
-        let node_path = query.bind(traversal_path.clone().rev().collect::<Vec<_>>())?;
+        let path = traversal_path.clone().rev().collect::<Vec<_>>();
+        let path: serde_json::Value = path.into();
+        let node_path = query.bind(path)?;
+        let created = query.bind(created)?;
+
         let sub_query = format!(
-            "(SELECT * FROM {table} WHERE path = {node_path} AND created <= {created} ORDER BY created DESC LIMIT 1)",
+            "SELECT * FROM (SELECT * FROM {table} WHERE path = {node_path} AND created <= {created} ORDER BY created DESC LIMIT 1)",
         );
 
         sub_queries.push(sub_query);
         traversal_path.next();
     }
 
     let mut sql: String = sub_queries.join(" UNION ");
-    sql.push_str("ORDER BY path DESC");
+
+    sql = format!("SELECT * FROM ({sql}) as t ");
+
+    if cfg!(feature = "embedded-db") {
+        sql.push_str("ORDER BY length(t.path) DESC");
+    } else {
+        sql.push_str("ORDER BY t.path DESC");
+    }
```

Comment:
I gather this is because sqlite doesn't support ordering on JSONB? However, logically, doesn't ordering by the length do what we want anyways? So maybe it's simpler just to go with that in both backends

Reply:
Yeah, postgres has array_length() and sqlite has length() so still need to match the feature

Reply:
damn